packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -186,19 +245,7 @@ const SimpleFormIterator = props => {
                                 disableRemove
                             ) && (
                                 <span className={classes.action}>
-                                    <Button
-                                        className={classNames(
-                                            'button-remove',
-                                            `button-remove-${source}-${index}`
-                                        )}
-                                        size="small"
-                                        onClick={removeField(index)}
-                                    >
-                                        <CloseIcon
-                                            className={classes.leftIcon}
-                                        />
-                                        {translate('ra.action.remove')}
-                                    </Button>
+                                    {getRemoveButton(index)}
```

Comment:
I would replace this with:
```js
{cloneElement(removeButton, { onClick: handleRemoveButtonClick(removeButton.props.onClick) })}
```

Where `handleRemoveButtonClick` is a function accepting the original remove button onClick handler and return another function, something like:

```js
const handleRemoveButtonClick = originalOnClickHandler = event => {
     removeField(index);
     if (originalOnClickHandler) {
        originalOnClickHandler(event);
     }
}
```